test/e2e/case5_policy_automation_test.go

```diff
@@ -135,7 +136,448 @@ var _ = Describe("Test policy automation", func() {
 				context.TODO(), "create-service-now-ticket", metav1.GetOptions{},
 			)
 			Expect(err).To(BeNil())
-			Expect(policyAutomation.Object["spec"].(map[string]interface{})["mode"]).To(Equal("disabled"))
+			Expect(
+				policyAutomation.Object["spec"].(map[string]interface{})["mode"],
+			).To(Equal(string(policyv1beta1.Disabled)))
+		})
+		// Create two events then two ansiblejobs for each event
+		It("Test mode = everyEvent without delayAfterRunSeconds", func() {
+			By("Patching policy to make both clusters back to Compliant")
+			opt := metav1.ListOptions{
+				LabelSelector: common.RootPolicyLabel + "=" + testNamespace + "." + case5PolicyName,
+			}
+			replicatedPlcList := utils.ListWithTimeout(clientHubDynamic, gvrPolicy, opt, 2, true, defaultTimeoutSeconds)
+			for _, replicatedPlc := range replicatedPlcList.Items {
+				replicatedPlc.Object["status"] = &policiesv1.PolicyStatus{
+					ComplianceState: policiesv1.Compliant,
+				}
+				_, err := clientHubDynamic.Resource(gvrPolicy).Namespace(replicatedPlc.GetNamespace()).UpdateStatus(
+					context.TODO(), &replicatedPlc, metav1.UpdateOptions{},
+				)
+				Expect(err).To(BeNil())
+			}
+			By("Waiting until the status on each cluster is updated to Compliant")
+			replicatedPlcList = utils.ListWithTimeout(clientHubDynamic, gvrPolicy, opt, 2, true, defaultTimeoutSeconds)
+			for _, replicatedPlc := range replicatedPlcList.Items {
+				Eventually(func() interface{} {
+					status := replicatedPlc.Object["status"]
+
+					return status.(map[string]interface{})["compliant"]
+				}, 30, 1).Should(Equal(string(policiesv1.Compliant)))
+			}
+
+			By("Patching policyAutomation with mode=everyEvent")
+			policyAutomation, err := clientHubDynamic.Resource(gvrPolicyAutomation).Namespace(testNamespace).Get(
+				context.TODO(), "create-service-now-ticket", metav1.GetOptions{},
+			)
+			Expect(err).To(BeNil())
+			policyAutomation.Object["spec"].(map[string]interface{})["mode"] = string(policyv1beta1.EveryEvent)
+			_, err = clientHubDynamic.Resource(gvrPolicyAutomation).Namespace(testNamespace).Update(
+				context.TODO(), policyAutomation, metav1.UpdateOptions{},
+			)
+			Expect(err).To(BeNil())
+
+			By("Should not create any new ansiblejob when Compliant, just one ansiblejob from once-mode test")
+			Consistently(func() interface{} {
```

Comment:
Can you delete the Ansible job in the once mode test so that it doesn't affect this test?